### PR TITLE
fix(inputsOutputs) to work with delete-confirm

### PR DESCRIPTION
**Pull request by LironHazan**

### CODE REVIEW

File: spec/inputsOutputs/inputsOutputs.spec.js [hunk 1]
```diff
@@ -126,7 +133,7 @@ describe('inputsOutputs page', function() {
       components.definitions.page.clickEnterBtn();
       components.definitions.page.saveInlineTypeElement();//save inline type
 
-      browser.sleep(200).then(done);
+      browser.sleep(2000).then(done);
```
Comment (edenpessach) [inline spec/inputsOutputs/inputsOutputs.spec.js]: Why did you changed it to 2000 instead of 200? is there a meaning to this?

File: spec/inputsOutputs/inputsOutputs.spec.js [hunk 2]
```diff
@@ -136,20 +143,18 @@ describe('inputsOutputs page', function() {
       expect(components.layout.isElementDisplayed(element(by.css('.popover')))).toBe(true);//popover should show up
 
       components.popovers.popover.clickNo();//click no btn
-      expect(components.layout.isElementDisplayed(element(by.css('.popover')))).toBe(false);//popover should hide
```
Comment (edenpessach) [inline spec/inputsOutputs/inputsOutputs.spec.js]: why remove this? popover should hide when clicking 'no'.

File: spec/inputsOutputs/inputsOutputs.spec.js [hunk 3]
```diff
@@ -163,7 +168,6 @@ describe('inputsOutputs page', function() {
       expect(components.layout.isElementDisplayed(element(by.css('.popover')))).toBe(true);//popover should show up
 
       components.popovers.popover.clickYes();//push 'Yes' btn
-      expect(components.layout.isElementPresent(element(by.css('.popover')))).toBe(false);//popover should be hide
```
Comment (edenpessach) [inline spec/inputsOutputs/inputsOutputs.spec.js]: why remove this? popover should hide when clicking 'yes'.

File: spec/inputsOutputs/inputsOutputs.spec.js [hunk 4]
```diff
@@ -136,20 +143,18 @@ describe('inputsOutputs page', function() {
...
       expect(components.inputsOutputs.page.countInputsOrOutputs(inputElement)).toBe(1);//input should exist
 
       components.layout.goToTopology();
-      components.topology.page.openNode();
-
-      expect(element.all(by.css('.propsContainer .propData')).get(0).getAttribute('value')).toBe('{"get_input":"name"}');//text should exist
 
-      browser.sleep(200);
+      components.topology.page.openNode();
+      components.topology.page.setPropertyValue('{"get_input":"name"}');
```
Comment (edenpessach) [inline spec/inputsOutputs/inputsOutputs.spec.js]: Why add this? didn't this test passed before?

File: spec/inputsOutputs/inputsOutputs.spec.js [hunk 5]
```diff
@@ -136,20 +143,18 @@ describe('inputsOutputs page', function() {
...
+      expect(element.all(by.css('.propertiesContainer .propItem .propData')).get(0).getAttribute('value')).toBe('{"get_input":"name"}');//text should exist
 
       components.layout.goToDefinitions();
 
       components.definitions.page.clickEditButton();//click edit button
-
+      components.definitions.page.showHiddenInputs();//click on field to show hidden inputs
```
Comment (edenpessach) [inline spec/inputsOutputs/inputsOutputs.spec.js]: Why add this? didn't this test passed before?

**Comment by LironHazan** [comment]
No, changed for debug purposes and forgot to revert.. Will change it back

File: spec/inputsOutputs/inputsOutputs.spec.js (hunk 1 above)
Comment (LironHazan) [inline spec/inputsOutputs/inputsOutputs.spec.js]: for debug, will change it back
.

File: spec/inputsOutputs/inputsOutputs.spec.js (hunk 2 above)
Comment (LironHazan) [inline spec/inputsOutputs/inputsOutputs.spec.js]: it's not a good test, throws error when couldn't find the element

File: spec/inputsOutputs/inputsOutputs.spec.js (hunk 4 above)
Comment (LironHazan) [inline spec/inputsOutputs/inputsOutputs.spec.js]: no it failed

File: spec/inputsOutputs/inputsOutputs.spec.js (hunk 5 above)
Comment (LironHazan) [inline spec/inputsOutputs/inputsOutputs.spec.js]: failed in front of my branch with the fix so I fixed it.

File: spec/inputsOutputs/inputsOutputs.spec.js (hunk 3 above)
Comment (LironHazan) [inline spec/inputsOutputs/inputsOutputs.spec.js]: it's not a good test, throws error when couldn't find the element

File: spec/inputsOutputs/inputsOutputs.spec.js (hunk 2 above)
Comment (edenpessach) [inline spec/inputsOutputs/inputsOutputs.spec.js]: we should test clicking no, replace "isElementDisplayed" with protractor isPresent method.

File: spec/inputsOutputs/inputsOutputs.spec.js (hunk 3 above)
Comment (edenpessach) [inline spec/inputsOutputs/inputsOutputs.spec.js]: we should test clicking no, replace "isElementDisplayed" with protractor isPresent method.

File: src/components/inputsOutputs/InputsOutputsPage.js [hunk 6]
```diff
@@ -12,7 +12,7 @@ exports.submitInputOrOutput = function(el){
   return el.$('[ng-click="addField()"]').click();
 };
 exports.deleteInputOrOutput = function(el){
-  return el.$('[ng-click="checkRemoveField(key,field)"]').click();
+  return el.$('[delete-confirm="checkRemoveField(key,field)"]').click();
```
Comment (edenpessach) [inline src/components/inputsOutputs/InputsOutputsPage.js]: since function is renamed we should change this as well.
checkRemoveField -> removeField

File: spec/inputsOutputs/inputsOutputs.spec.js (hunk 5 above)
Comment (edenpessach) [inline spec/inputsOutputs/inputsOutputs.spec.js]: remove all changes of this test, it fails just because "isElementDisplayed".
 replace with protractor isPresent method

File: spec/inputsOutputs/inputsOutputs.spec.js [hunk 7]
```diff
@@ -133,23 +142,23 @@ describe('inputsOutputs page', function() {
       components.layout.goToInputsOutputs();
 
       components.inputsOutputs.page.deleteInputOrOutput(inputElement, 'input');// try to delete input
-      expect(components.layout.isElementDisplayed(element(by.css('.popover')))).toBe(true);//popover should show up
+      expect((element(by.css('.popover')).isDisplayed())).toBeTruthy();//popover should show up
 
       components.popovers.popover.clickNo();//click no btn
-      expect(components.layout.isElementDisplayed(element(by.css('.popover')))).toBe(false);//popover should hide
+      browser.sleep(200);
```
Comment (edenpessach) [inline spec/inputsOutputs/inputsOutputs.spec.js]: why is this here?

File: spec/inputsOutputs/inputsOutputs.spec.js [hunk 8]
```diff
@@ -133,23 +142,23 @@ describe('inputsOutputs page', function() {
...
+      expect(element(by.css('.popover')).isPresent()).toBe(false);
```
Comment (edenpessach) [inline spec/inputsOutputs/inputsOutputs.spec.js]: decide on one way to check for popover:
+      expect((element(by.css('.popover')).isDisplayed())).toBeTruthy();
+      expect(element(by.css('.popover')).isPresent()).toBe(false);

File: spec/inputsOutputs/inputsOutputs.spec.js (hunk 4 above)
Comment (edenpessach) [inline spec/inputsOutputs/inputsOutputs.spec.js]: still needs to remove this

File: spec/inputsOutputs/inputsOutputs.spec.js (hunk 5 above)
Comment (edenpessach) [inline spec/inputsOutputs/inputsOutputs.spec.js]: still needs to remove this

File: spec/inputsOutputs/inputsOutputs.spec.js [hunk 9]
```diff
@@ -160,10 +169,9 @@ describe('inputsOutputs page', function() {
       components.layout.goToInputsOutputs();
 
       components.inputsOutputs.page.deleteInputOrOutput(inputElement, 'input');// try to delete input
-      expect(components.layout.isElementDisplayed(element(by.css('.popover')))).toBe(true);//popover should show up
+      expect((element(by.css('.popover')).isDisplayed())).toBeTruthy();//popover should show up
 
       components.popovers.popover.clickYes();//push 'Yes' btn
-      expect(components.layout.isElementPresent(element(by.css('.popover')))).toBe(false);//popover should be hide
```
Comment (edenpessach) [inline spec/inputsOutputs/inputsOutputs.spec.js]: why remove this? we still need to check popover disappeared after pressing yes

File: spec/inputsOutputs/inputsOutputs.spec.js [hunk 10]
```diff
@@ -72,13 +71,17 @@ describe('inputsOutputs page', function() {
...
   });
+
   describe('remove input and output', function() {
 
     it('should remove input', function(done) {
       // remove input
       components.inputsOutputs.page.setInputOrOutputFields(inputElement, 'input');//fill fields
       components.inputsOutputs.page.submitInputOrOutput(inputElement);//add
       components.inputsOutputs.page.deleteInputOrOutput(inputElement, 'input');//remove element
+      expect((element(by.css('.popover')).isDisplayed())).toBeTruthy();//popover should show up
```
Comment (edenpessach) [inline spec/inputsOutputs/inputsOutputs.spec.js]: move checking is popover present to a component, this code is repeated many times throughout the code.

File: spec/inputsOutputs/inputsOutputs.spec.js (hunk 5 above)
Comment (LironHazan) [inline spec/inputsOutputs/inputsOutputs.spec.js]: removed

File: spec/inputsOutputs/inputsOutputs.spec.js (hunk 4 above)
Comment (LironHazan) [inline spec/inputsOutputs/inputsOutputs.spec.js]: removed
